src/base/bittorrent/torrent.h

```diff
@@ -92,6 +94,8 @@ namespace BitTorrent
 
     uint qHash(TorrentState key, uint seed);
 
+    using TagsSet = OrderedSet<QString, Utils::Compare::NaturalLessThan<Qt::CaseInsensitive>>;
```

Comment:
During my last testing, I found a serious breaking change: the new `TagsSet` will determine element uniqueness case-insensitively, this is different than before which is case-sensitive.

Would we allow this breaking change in the v4.4.x branch? Personally I don't quite like it but I'm still asking because working around this doesn't seem to be easy...

Reply:
Just to be sure... We can't make them case sensitive, can we?

Reply:
>We can't make them case sensitive, can we?

We can but then there is no point (no performance advantages) in using `OrderedSet<QString>`, we still have to convert it to list type and sort case-insensitively.

Reply:
>we still have to convert it to list type and sort case-insensitively.

I mean "Can't we sort them case-sensitively?"

Reply:
Otherwise, it looks like we'll have to go back to where we started, i.e. either:
1. Store tags in QSet and produce/sort list every time we need them sorted,
2. Store tags in sorted list and produce QSet every time we need "set-related calculations" (or perform such calculations slowly with lists themselves),
3. Define custom SortedSet class and store items inside it in both `list` and `set` at the same time.

Reply:
>I mean "Can't we sort them case-sensitively?"

Seems this problem can be solved by introducing sub-comparison. PR updated.